src/printer.js

```diff
@@ -2265,7 +2265,8 @@ function printBinaryishExpressions(path, parts, print) {
       parts.push(path.call(print, "left"));
     }
 
-    parts.push(" ", node.operator, line, path.call(print, "right"));
+    const allowLine = !namedTypes.Literal.check(node.right);
```

Comment:
I just remembered that doing this type of check is better than `node.type === "Literal"`. It handles different across parsers better, i.e. this works on both and babylon even though they handle the literal types differently.